File: src/main/java/sunshine/common/ErrorResponse.java
package sunshine.common;

import java.util.Map;

public record ErrorResponse(String code, String message, Map<String, Object> details) {
    public static ErrorResponse of(ErrorCode code, String message, Map<String, Object> details) {
        return new ErrorResponse(
            code.name(),
            message != null ? message : code.getDefaultMessage(),
            details
        );
    }
}

File: src/main/kotlin/sunshine/common/ErrorResponse.kt
package sunshine.common

/**
 * 에러 응답을 표준화하는 레코드.
 */
data class ErrorResponse(
    val code: String,
    val message: String,
    val details: Map<String, Any>?
) {
    companion object {
        fun of(code: ErrorCode, message: String?, details: Map<String, Any>?): ErrorResponse =
            ErrorResponse(
                code.name,
                message ?: code.defaultMessage,
                details
            )
    }
}
